livekit-agents/livekit/agents/multimodal/multimodal_agent.py

```diff
@@ -134,12 +142,30 @@ def start(
         self._session = self._model.session(
             chat_ctx=self._chat_ctx, fnc_ctx=self._fnc_ctx
         )
-        self._main_atask = asyncio.create_task(self._main_task())
+
+        # Create a task to wait for initialization and start the main task
+        async def _init_and_start():
+            try:
+                await self._session._init_sync_task
```

Comment:
Weirdly, I was testing out this branch using the virtual assistant sandbox playground had an issue in which the agent wouldn't start, while waiting for this task to complete. Does that make sense at all? I removed it temporarily from my local `venv` and the agent works fine, albiet after a 2 second waiting period to sync the initial context.

Reply:
Can you share the package and the version you installed with the issue? I can take a look and try to reproduce it.

Reply:
Using the latest commit actually helped fix the issue, apologies! Thanks for the tip that it could've been an issue with the commit I was on.